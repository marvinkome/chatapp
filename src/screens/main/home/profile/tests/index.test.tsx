import * as React from 'react';
import { create } from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/test-utils';
import wait from 'waait';
import { Profile } from '../index';
import View from '../view';
import query from '../view/gql';

const mocks = [
    {
        request: {
            query
        },
        result: {
            data: {
                user: {
                    id: '1',
                    username: 'johndoe',
                    about: 'Hey there'
                }
            }
        }
    }
];

describe('<Profile/>', () => {
    it('renders correctly', () => {
        const comp = create(
            <MockedProvider mocks={mocks} addTypename={false}>
                <Profile />
            </MockedProvider>
        );
        expect(comp.toJSON()).toMatchSnapshot();
    });

    // it should show user avatar, display name and status
    it('should show user avatar, display name and status', async () => {
        const comp = create(
            <MockedProvider mocks={mocks} addTypename={false}>
                <View />
            </MockedProvider>
        );
        await wait(0);
        const wrapper = comp.root;
        const userDataView = wrapper.findByProps({
            ['data-testId']: 'user-data-view'
        });

        // assert it renders an avatar component
        const avatarComponent = userDataView.findAllByProps({ rounded: true });
        expect(avatarComponent).toBeTruthy();

        // asser it displays user name
        const texts = userDataView.findAllByType('Text');
        const name = texts[0].children[0];

        expect(name).toBe('johndoe');

        // assert it displays status
        const status = texts[1].children[0];
        expect(status).toBe('Hey there');
    });

    // it should have edit profile when image is clicked

    it('should render settings items', async () => {
        const comp = create(
            <MockedProvider mocks={mocks} addTypename={false}>
                <View />
            </MockedProvider>
        );
        await wait(0);

        const wrapper = comp.root;
        const settingsComp = wrapper.findByProps({ ['data-testId']: 'list' });

        // should have 4 children
        const listItem = settingsComp.findAll((node) => node.props.hideChevron);
        expect(listItem).toHaveLength(5);

        // list 4 options
        // array of children text
        const titleLists = listItem.map((item) => item.props.title);
        expect(titleLists).toEqual([
            'Invite Friends',
            'FAQ',
            'Help',
            'About',
            'Logout'
        ]);
    });
});
