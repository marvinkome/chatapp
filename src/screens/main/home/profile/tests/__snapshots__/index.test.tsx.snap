// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    data-testId="list"
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#bbb",
          "borderTopWidth": 1,
          "marginTop": 20,
        },
        Object {
          "borderTopWidth": 0,
        },
      ]
    }
  >
    <View
      disabled={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "#bbb",
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
          Object {
            "borderBottomColor": "hsl(0, 0%, 90%)",
            "paddingBottom": 15,
            "paddingTop": 15,
          },
          false,
          false,
        ]
      }
      underlayColor="white"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginLeft": 10,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#43484d",
                      "fontSize": 17.5,
                    },
                    Object {
                      "marginLeft": 10,
                    },
                    Object {
                      "color": "hsl(0, 0%, 45%)",
                      "fontFamily": "Karla",
                      "fontSize": 15,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Invite Friends
            </Text>
          </View>
          <View>
            <View />
          </View>
        </View>
      </View>
    </View>
    <View
      disabled={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "#bbb",
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
          Object {
            "borderBottomColor": "hsl(0, 0%, 90%)",
            "paddingBottom": 15,
            "paddingTop": 15,
          },
          false,
          false,
        ]
      }
      underlayColor="white"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginLeft": 10,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#43484d",
                      "fontSize": 17.5,
                    },
                    Object {
                      "marginLeft": 10,
                    },
                    Object {
                      "color": "hsl(0, 0%, 45%)",
                      "fontFamily": "Karla",
                      "fontSize": 15,
                    },
                    undefined,
                  ],
                ]
              }
            >
              FAQ
            </Text>
          </View>
          <View>
            <View />
          </View>
        </View>
      </View>
    </View>
    <View
      disabled={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "#bbb",
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
          Object {
            "borderBottomColor": "hsl(0, 0%, 90%)",
            "paddingBottom": 15,
            "paddingTop": 15,
          },
          false,
          false,
        ]
      }
      underlayColor="white"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginLeft": 10,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#43484d",
                      "fontSize": 17.5,
                    },
                    Object {
                      "marginLeft": 10,
                    },
                    Object {
                      "color": "hsl(0, 0%, 45%)",
                      "fontFamily": "Karla",
                      "fontSize": 15,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Help
            </Text>
          </View>
          <View>
            <View />
          </View>
        </View>
      </View>
    </View>
    <View
      disabled={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "#bbb",
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
          Object {
            "borderBottomColor": "hsl(0, 0%, 90%)",
            "paddingBottom": 15,
            "paddingTop": 15,
          },
          false,
          false,
        ]
      }
      underlayColor="white"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginLeft": 10,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#43484d",
                      "fontSize": 17.5,
                    },
                    Object {
                      "marginLeft": 10,
                    },
                    Object {
                      "color": "hsl(0, 0%, 45%)",
                      "fontFamily": "Karla",
                      "fontSize": 15,
                    },
                    undefined,
                  ],
                ]
              }
            >
              About
            </Text>
          </View>
          <View>
            <View />
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "#bbb",
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
          Object {
            "borderBottomColor": "hsl(0, 0%, 90%)",
            "paddingBottom": 15,
            "paddingTop": 15,
          },
          false,
          false,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginLeft": 10,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#43484d",
                      "fontSize": 17.5,
                    },
                    Object {
                      "marginLeft": 10,
                    },
                    Object {
                      "color": "hsl(0, 0%, 45%)",
                      "fontFamily": "Karla",
                      "fontSize": 15,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Logout
            </Text>
          </View>
          <View>
            <View />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
